### PR TITLE
Add rewards table to dashboard page

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export function FilterData() {
+  const { name } = useProjectRewardsFilterDataSidePanel();
+  const { Panel } = useSidePanel({ name });
+  const { filters, setFilters, saveFilters, resetFilters } = useFilterData<RewardsTableFilters>();
+
+  return (
+    <Panel>
+      <SidePanelHeader
+        title={{
+          children: <Translate token={"manageProjects:detail.filters.titles.rewards"} />,
```
Comment (haydencleary) [inline app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx:24]: You're using the manage projects translations

File: app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        }}
+        canGoBack={false}
+        canClose={true}
+      />
+      <SidePanelBody>
+        <ContributorProjectFilter
+          selectedUser={filters.contributors?.map(id => id.toString())}
```
Comment (haydencleary) [inline app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx:31]:
```suggestion
          selectedUser={filters.contributors?.map(String)}
```

File: app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          onSelect={(users: string[]) => {
+            setFilters({ contributors: users.map(user => Number(user)) });
```
Comment (haydencleary) [inline app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx:33]:
```suggestion
            setFilters({ contributors: users.map(Number) });
```

File: app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        <CurrencyFilter selectedCurrencies={filters.currencies} onSelect={currencies => setFilters({ currencies })} />
+      </SidePanelBody>
+      <SidePanelFooter>
+        <div className={"flex w-full flex-row items-center justify-end gap-lg"}>
+          <Button size={"md"} variant={"secondary"} onClick={() => resetFilters()}>
+            <Translate token={"manageProjects:detail.filters.reset"} />
+          </Button>
+
+          <Button size={"md"} variant={"secondary"} onClick={() => saveFilters()}>
+            <Translate token={"manageProjects:detail.filters.save"} />
```
Comment (haydencleary) [inline app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx:45]: More manage projects translations, I added these to common.form translations

File: app/my-dashboard/_features/rewards-table/rewards-table.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  const [filters, setFilters] = useState<RewardsTableFilters>({});
+  const { githubUserId } = useAuthUser();
+
+  const { columns, selectedIds, setSelectedIds, sorting, setSorting, sortingParams } = useFilterColumns();
+
+  const queryParams: Partial<GetRewardsQueryParams> = {
+    search: debouncedSearch,
+    ...filters,
+    ...sortingParams,
+    recipientIds: [githubUserId ?? 0],
```
Comment (haydencleary) [inline app/my-dashboard/_features/rewards-table/rewards-table.tsx:43]:
```suggestion
    recipientIds: githubUserId ? [githubUserId] : undefined,
```

Wouldn't this be safer ?

**Review by haydencleary — CHANGES_REQUESTED** [review]

File: app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx (hunk 1 above)
Comment (alexbeno) [reply app/my-dashboard/_features/rewards-table/_components/filter-data/filter-data.tsx:24]: filter are not done yet but a can put the good translation

**Review by haydencleary — APPROVED** [review]

**Pull request merged**